assignment/core/src/main/java/com/assignment/core/schedulers/JobScheduler.java

```diff
@@ -0,0 +1,55 @@
+package com.assignment.core.schedulers;
+
+import com.assignment.core.services.JobTriggerService;
+import org.apache.sling.commons.scheduler.ScheduleOptions;
+import org.apache.sling.commons.scheduler.Scheduler;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The  JobScheduler class is a scheduled task that triggers a job at a fixed interval.
+ * It implements the  Runnable interface to be used by the Sling Scheduler.
+ */
+@Component(service = Runnable.class, immediate = true)
+public class JobScheduler implements Runnable {
+
+    private static final Logger LOG = LoggerFactory.getLogger(JobScheduler.class);
+
+    @Reference
+    private Scheduler scheduler;
+
+    @Reference
+    private JobTriggerService jobTriggerService;
+
+    /**
+     * Executes the scheduled job.
+     */
+    @Override
+    public void run() {
+        LOG.info("Scheduler is triggering job...");
+        jobTriggerService.trigger();
```

Comment:
No need to make separate class service for adding and triggering job, perform it in scheduler class itself.